### PR TITLE
[WIP] Upgrade guide BDB v0.10 -> v1.0

### PR DESCRIPTION
**Pull request by TimDaub**
WIP, @r-marques and me will add notes on all breaking changes.

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -1,23 +1,32 @@
...
 # How to Generate the HTML Version of the Long-Form Documentation
 
-If you want to generate the HTML version of the long-form documentation on your local machine, you need to have Sphinx and some Sphinx-contrib packages installed. To do that, go to a subdirectory of `docs` (e.g. `docs/server`) and do:
-```bash
-pip install -r requirements.txt
-```
+If you want to generate the HTML version of the long-form documentation on your
+local machine, you need to have Sphinx and some Sphinx-contrib packages
+installed. To do that, go to a subdirectory of `docs` (e.g. `docs/server`) and
+do: ```bash pip install -r requirements.txt ```
```
Comment (ttmc) [inline docs/README.md]: What happened here? Did some program come through and force all the lines (in a .md file) to be less than some max line length?

Anyway, the end result is that this line doesn't work any more, at least not on GitHub.

File: docs/README.md [hunk 2]
```diff
@@ -1,23 +1,32 @@
...
 
 You can then generate the HTML documentation _in that subdirectory_ by doing:
-```bash
-make html
-```
+```bash make html ```
```
Comment (ttmc) [inline docs/README.md]: This also doesn't work any more. The "backtick backtick backtick bash" should be on its own line.

File: docs/upgrade-guides/v0.10-->v1.0.md [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+[big.js](https://github.com/MikeMcl/big.js)) when dealing with large numbers in
+Javascript. Additionally, numbers are capped at 9e18 to stay comfortably within
+the boundary of a 64 bit signed integer.
+
+\* Try this in the Chrome developer console: `2**60 == 2**60+1`.
+
+
+#### Input `fulfills.txid` is now `transaction_id`
+
+We renamed a TRANSFER transaction's `inputs.fulfills.txid` to
```
Comment (ttmc) [inline docs/upgrade-guides/v0.10-->v1.0.md]: Isn't this just one example of 

`txid` and `tx_id` becomes `transaction_id`

(which was already mentioned).

File: docs/README.md (hunk 1 above)
Comment (TimDaub) [inline docs/README.md]: :/ OK, I'll revert that

File: docs/upgrade-guides/v0.10-->v1.0.md (hunk 3 above)
Comment (TimDaub) [inline docs/upgrade-guides/v0.10-->v1.0.md]: Yup, I decided to go over the changes in more detail.

See that there are now sections for e.g. all HTTP endpoints that changed

File: docs/upgrade-guides/v0.10-->v1.0.md [hunk 4]
```diff
@@ -0,0 +1,396 @@
...
+
+
+Changes:
+
+- All notion of `_links` was removed. In case of querying the status of a
+  transaction already included in a block marked `VALID`, no `_links` object is
+  provided anymore. The response object now only contains a single key value
+  pair named `status`
+- The query parameter `tx_id` was renamed to `transaction_id`, e.g. `GET
+  /api/v1/statuses?transaction_id=`
```
Comment (ttmc) [inline docs/upgrade-guides/v0.10-->v1.0.md]: Don't you mean `GET /api/v1/statuses?transaction_id={transaction_id}`?

File: docs/upgrade-guides/v0.10-->v1.0.md [hunk 5]
```diff
@@ -0,0 +1,396 @@
...
+
+Changes:
+
+- The query parameter `tx_id` was renamed to `transaction_id`, e.g. `GET
+  /api/v1/blocks?transaction_id`
+
+
+## Breaking Changes to the WebSocket Event Stream API
+
+In the event object sent to a listener, `tx_id` was renamed to
```
Comment (ttmc) [inline docs/upgrade-guides/v0.10-->v1.0.md]: What about the name of the WebSocket Event Stream API endpoint (was `/api/v1/streams/valid_tx`)? Is the `tx` now `transaction`?

File: docs/upgrade-guides/v0.10-->v1.0.md (hunk 4 above)
Comment (TimDaub) [inline docs/upgrade-guides/v0.10-->v1.0.md]: I was consistently (throughout the document) lazy, but I can add that if you want

File: docs/upgrade-guides/v0.10-->v1.0.md (hunk 5 above)
Comment (TimDaub) [inline docs/upgrade-guides/v0.10-->v1.0.md]: Good point, I'll add.

**Comment by TimDaub** [comment]
@sbellem, there is a section on CC V2 [here](https://github.com/bigchaindb/bigchaindb/pull/1547/files#diff-747c350ebd24108cad93aae8e480cf31R91). You might want to check it out.

**Comment by r-marques** [comment]
Added a note about the change in the transaction version.

File: docs/upgrade-guides/v0.10-->v1.0.md [hunk 6]
```diff
@@ -0,0 +1,401 @@
...
+For all future releases, we commit to not introduce breaking changes to the
+public interfaces of BigchainDB's:
+
+- [Data
+  model](https://docs.bigchaindb.com/projects/server/en/latest/data-models/index.html)
+- [HTTP
+  API](https://docs.bigchaindb.com/projects/server/en/latest/http-client-server-api.html)
+- [WebSocket Event Stream
+  API](https://docs.bigchaindb.com/projects/server/en/latest/websocket-event-stream-api.html)
+- [official Python driver](https://github.com/bigchaindb/bigchaindb-driver)
```
Comment (TimDaub) [inline docs/upgrade-guides/v0.10-->v1.0.md]: Remove

File: docs/upgrade-guides/v0.10-->v1.0.md [hunk 7]
```diff
@@ -0,0 +1,401 @@
...
+- [official Javascript
```
Comment (TimDaub) [inline docs/upgrade-guides/v0.10-->v1.0.md]: remove

**Comment by r-marques** [comment]
Note: Update the `Inputs` section if #1596 goes through

**Comment by r-marques** [comment]
@TimDaub I think we can merge this. All the breaking changes are in the document.

**Pull request merged**